components/draw.js

```diff
@@ -45,38 +43,27 @@ function ConvergameDraw() {
 	      img.src = imagePath;
 	    }
 	  };
-
-  this.circle = function(centreX, centreY, radius, style)
-  {
-    this.convergame.ctx.strokeStyle = style;
-    this.convergame.ctx.beginPath();
-    this.convergame.ctx.arc(this.prepNum(centreX * this.getXScale()), this.prepNum(centreY * this.getYScale()), this.prepNum(radius * this.getXScale()), 0, 2*Math.PI);
-    this.convergame.ctx.stroke();
-  };
-
-  this.filledCircle = function(centreX, centreY, radius, strokeStyle, fillStyle)
+  this.circle = function(centreX, centreY, radius, strokeStyle, fillStyle )
   {
     this.convergame.ctx.strokeStyle = strokeStyle;
-    this.convergame.ctx.fillStyle = fillStyle;
+    if(typeof(fillStyle) != 'undefined') {this.convergame.ctx.fillStyle = fillStyle}
     this.convergame.ctx.beginPath();
     this.convergame.ctx.arc(this.prepNum(centreX * this.getXScale()), this.prepNum(centreY * this.getYScale()), this.prepNum(radius * this.getXScale()), 0, 2*Math.PI);
     this.convergame.ctx.stroke();
-    this.convergame.ctx.fill();
-  };
-
-  this.rectangle = function(x, y, width, height, style)
-  {
-    this.convergame.ctx.strokeStyle = style;
-    this.convergame.ctx.strokeRect(this.prepNum(x*this.getXScale()), this.prepNum(y * this.getYScale()), this.prepNum(width*this.getXScale()), this.prepNum(height*this.getYScale()));
+    if(typeof(fillStyle) != 'undefined') {this.convergame.ctx.fill();}
   };
 
-  this.filledRectangle = function(x, y, width, height, strokeStyle, fillStyle)
+  this.rectangle = function(x, y, width, height, strokeStyle, fillStyle)
   {
-    this.convergame.ctx.strokeStyle = strokeStyle;
-    this.convergame.ctx.fillStyle = fillStyle;
-    this.convergame.ctx.fillRect(this.prepNum(x*this.getXScale()), this.prepNum(y * this.getYScale()), this.prepNum(width*this.getXScale()), this.prepNum(height*this.getYScale()));
+    if(typeof(fillStyle) != 'undefined') {
```

Comment:
Instead of this if, perhaps just use `ctx.rect` followed by `ctx.stroke` and then an optional `ctx.fill`?